TGM/src/main/java/network/warzone/tgm/util/menu/KitEditorMenu.java

```diff
@@ -0,0 +1,188 @@
+package network.warzone.tgm.util.menu;
+
+import lombok.Getter;
+import network.warzone.tgm.TGM;
+import network.warzone.tgm.modules.kit.Kit;
+import network.warzone.tgm.modules.kit.KitNode;
+import network.warzone.tgm.modules.kit.types.ItemKitNode;
+import network.warzone.tgm.util.itemstack.ItemFactory;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryAction;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.event.inventory.InventoryDragEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Created by Atdit on 24/04/2020
+ */
+public class KitEditorMenu implements Listener {
+    @Getter private final Inventory inventory;
+
+    @Getter
+    private Kit kit;
+
+    private final HashMap<Integer, MenuAction> actions = new HashMap<>();
+
+    public KitEditorMenu(List<Kit> kits, String mapName) {
+        String name = ChatColor.UNDERLINE + "Kit Editor:" + ChatColor.RESET + " " + mapName;
+        int slots = 5 * 9;
+
+        this.inventory = Bukkit.createInventory(null, slots, name);
+        TGM.registerEvents(this);
+
+        kit = kits.get(0); // Only runs if there is one kit
```

Comment:
Why wouldn't there be a kit? What happens if that's the case?

Reply:
The comment is there in case anyone is wondering what happens if there is no kit or more than one kit in the map json. In that case, the code won't be reached as it's already determined in the KitLoaderModule how many kits there are. Also, it won't run if there are zero kits since nothing can be edited, obviously. Afaik there currently aren't any maps with no kit in the repository but I can imagine someone making maps where you don't get any kit at all. It could be useful for a sumo wrestling map.

Reply:
The only thing I could think of is in the maps where the default TGM classes are used, since then I presume technically there aren’t any defined kits.

Reply:
@BennyDoesStuff yeah, about classes: I just tested it on a map with classes and the previous code in the KitLoaderModule wouldn't have run, resulting in the item appearing in the spectator inventory. It's fixed now. In the future, next to adding the ability to edit the kit layout when several kits are present, we could also change it to work for maps with classes.